packages/ui/src/components/va-form/VaForm.stories.ts

```diff
@@ -1,12 +1,443 @@
+        Validate
+      </va-button>
+      <va-button @click="$refs.validateAndResetValidationForm.resetValidation()">
+        Reset validation
+      </va-button>
+    </div>
+  `,
+})
+
+ValidateAndResetValidation.play = async ({ canvasElement, step }) => {
+  const canvas = within(canvasElement)
+  const firstInput = canvas.getByRole('textbox', { name: 'prop is not reset' }) as HTMLElement
+  const secondInput = canvas.getByRole('textbox', { name: 'validated rule is reset' }) as HTMLElement
+  const validateButton = canvas.getByRole('button', { name: 'Validate' }) as HTMLElement
+  const resetButton = canvas.getByRole('button', { name: 'Reset validation' }) as HTMLElement
+
+  await step('Validates input vith error', async () => {
+    await userEvent.click(validateButton)
+    expect(secondInput.getAttribute('aria-invalid')).toEqual('true')
+    expect(firstInput.getAttribute('aria-invalid')).toEqual('false')
+  })
+
+  await step('Reset inputs validation', async () => {
+    await userEvent.click(resetButton)
+    expect(secondInput.getAttribute('aria-invalid')).toEqual('false')
+    expect(firstInput.getAttribute('aria-invalid')).toEqual('false')
+  })
+}
+
+export const Reset = () => ({
+  components: { VaForm, VaInput, VaButton },
+  data: () => ({ username: 'root', password: 'admin' }),
+  template: `
+    <va-form
+      class="flex flex-col max-w-sm"
+      tag="form"
+      ref="resetForm"
+      immediate
+    >
+      <va-input
+        v-model="username"
+        label="Username"
+      />
+      <va-input
+        class="mt-2"
+        v-model="password"
+        label="Password"
+      />
+      <va-input
+        class="mt-2"
+        label="Error"
+        :rules="[() => 'error']"
+      />
+      <va-button 
+        class="mt-2"
+        type="submit" 
+        @click.prevent="$refs.resetForm.reset()"
+      >
+        Reset inputs and validation
+      </va-button>
+    </va-form>
+  `,
+})
+
+Reset.play = async ({ canvasElement, step }) => {
+  const canvas = within(canvasElement)
+  const firstInput = canvas.getByRole('textbox', { name: 'Username' }) as HTMLElement
+  const secondInput = canvas.getByRole('textbox', { name: 'Password' }) as HTMLElement
+  const thirdInput = canvas.getByRole('textbox', { name: 'Error' }) as HTMLElement
+  const button = canvas.getByRole('button', { name: 'Reset inputs and validation' }) as HTMLElement
+
+  await step('Resets form', async () => {
+    await userEvent.click(button)
+    expect(firstInput).toHaveValue('')
+    expect(secondInput).toHaveValue('')
+    expect(thirdInput.getAttribute('aria-invalid')).toEqual('false')
+  })
+}
+
+export const FormSubmit = () => ({
+  components: { VaForm, VaInput, VaButton },
+  data: () => ({ isSubmit: false }),
+  template: `
+    <va-form
+      class="flex flex-col max-w-sm"
+      tag="form"
+      ref="formSubmit"
+      @submit.prevent="isSubmit = true"
+    >
+      <va-input
```

Comment:
Let's just have one input here and one button. Remove label, class, unnecessary rules, div at the bottom.